### ISSUE TITLE
Implement `.get` on Kubernetes resources

### ISSUE THREAD

**Issue by hausdorff** [pulumi/pulumi-kubernetes#231]
This is a critical blocker for adopting existing Kubernetes resources.

**Comment by joeduffy** [comment]
@hausdorff, I just ran into the need for this myself. Think it is coming soon?

**Comment by hausdorff** [comment]
This will make it for M18, I'm looking right now.

But, this turns out to not be as trivial as I'd thought. Our implementation of `Read` turns out to be built on the assumption that every invocation contains the object itself. In `.get()` this is not the case.

So, @joeduffy, @lukehoban, a key question is: is there any situation in which a normal user can be using a `>= v0.18.0` node SDK with a `< v0.18.0` provider?

If not, then we're in the clear, and the path is straightforward:

- [ ] Fix `Read` to be able to read using only `req.ID`
- [ ] Plumb that ID down into the client's read logic
- [ ] Write tests

This should be doable in a couple hours, I think.

**Comment by lukehoban** [comment]
> is there any situation in which a normal user can be using a >= v0.18.0 node SDK with a < v0.18.0 provider?

No.  The post install step on the `@pulumi/kubernetes` NPM package should install the matching version of the Kubernets provider binary.

**Comment by hausdorff** [comment]
lol @daveremy, did you mean to say more? :)

**Comment by daveremy** [comment]
Nope sorry, my phone seems to have a mind of its own, please disregard.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
